### ISSUE TITLE
Unable to compile model after loading model with GradientAccumulateOptimizer wrapper

### ISSUE THREAD

**Issue by andreped** [andreped/GradientAccumulator#42]
See [this](https://github.com/andreped/GradientAccumulator/blob/3a2eb5b46bb2a95dd29895e6ca494ab2299fca12/tests/test_optimizer_wrapper.py#L134) unit test to reproduce.

Basically, if we write `compile=True` after `load_model` [here](https://github.com/andreped/GradientAccumulator/blob/3a2eb5b46bb2a95dd29895e6ca494ab2299fca12/tests/test_optimizer_wrapper.py#L93), we need to provide the optimizer as a `custom_object`. This should not be necessary.

Perhaps some serialization is missing?

Error log:
```
ValueError: Unknown config_item: 'SGD'. Please ensure you are using a `keras.utils.custom_object_scope` and that
this object is included in the scope. See https://www.tensorflow.org/guide/keras/save_and_serialize#registering_the_custom_object for details.
```

**Comment by andreped** [comment]
We may have to do something like [this](https://github.com/keras-team/keras/blob/v2.12.0/keras/mixed_precision/loss_scale_optimizer.py#L873).

**Comment by andreped** [comment]
Fixed in 1595bd96be81db69e0fcc831b07b3116f1b16bc6.
